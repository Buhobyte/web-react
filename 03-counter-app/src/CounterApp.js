import React, { useState } from 'react'
import PropTypes from 'prop-types'

const CounterApp = ({ value = 100 }) => {

    // const handleAdd = (e) => console.log(e);
    const handleAdd = (e) => console.log(e);

    const handleAdd2 = (e) => {

        console.log('contenido funcion handleadd2() se ejecutar al renderizar elementos del componente')

        return (e) => {
            console.log(e);
        }
    }

    const [counter, setCounter] = useState(0); // retorna un array [], HOOK es una funcion q permite conectarse a caracteristicas de react

    const handleAdd3 = () => {
        // setCounter(counter + 1);
        setCounter((c) => c + 1);
    }

    // Ejercicio restart resetear y sumar
    const [count, setCount] = useState(value);

    const handleSubtract = () => {
        setCount(count - 1);
    }
    const handleReset = () => {
        setCount(value);
    }
    const handlePlus = () => {
        setCount(count + 1);
    }


    return (
        <>
            <h1>CounterApp</h1>
            <h1  >{counter}</h1>
            <button onClick={handleAdd3}> +1  </button>
            <button onClick={handleAdd}> +3 </button>
            <button onClick={handleAdd2()}> +2 </button>
            <br />
            <h1>Ejercicio count</h1>
            <h1 id="idCounter">{count}</h1>
            <button id="btnPlus" onClick={handlePlus}> +1  </button>
            <button id="btnReset" onClick={handleReset}> Reset  </button>
            <button id="btnSubtract" onClick={handleSubtract}> -1  </button>

        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number
}

export default CounterApp
