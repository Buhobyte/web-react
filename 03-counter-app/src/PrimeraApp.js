import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

// const PrimeraApp = (props) => {
// const PrimeraApp = ({ saludo="Valor por defecto" }) => {
const PrimeraApp = ({ saludo,subtitulo }) => {

    return (
        <>
            <h1> {saludo}</h1 >
            <h1> {subtitulo}</h1 >
            <p> Parrago de mi div</p>
        </>

    );
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired,
    subtitulo: PropTypes.string
}

PrimeraApp.defaultProps ={
    subtitulo: "Soy un sibtitulo"
}

export default PrimeraApp;