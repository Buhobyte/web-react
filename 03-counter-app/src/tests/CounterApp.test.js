import React from 'react';
// import { render } from '@testing-library/react';
import PrimeraApp from '../PrimeraApp';
import { shallow } from 'enzyme';
import CounterApp from '../CounterApp';

describe('Prueba de CounterApp', () => {

    // puedo dejarlo "let wrapper;" pero el inteligense de "visual code" no funciona
    let wrapper = shallow(<CounterApp />);

    // reiniciar cada test el componente
    beforeEach(() => {
        wrapper = shallow(<CounterApp />);
    })

    test('debe mostrar <CounterApp/> correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('debe mostrar el valor por defecto', () => {

        const wrapper = shallow(<CounterApp value={100} />);

        const valorCounter = wrapper.find('#idCounter').text().trim();

        expect(valorCounter).toBe("100");

    })

    test('deberia aumentar 1 al hacer click en el boton plus', () => {

        const valorCounter = parseInt(wrapper.find('#idCounter').text().trim()); //100
        wrapper.find('#btnPlus').simulate('click'); //+1
        const valueAfterClickPlus = parseInt(wrapper.find('#idCounter').text().trim()); //101

        const valueCounterNew = (valueAfterClickPlus - 1);
        expect(valorCounter).toBe(valueCounterNew);

    })


    test('deberia reducir 1 al hacer click en el boton btnSubtract', () => {

        const valorCounter = parseInt(wrapper.find('#idCounter').text().trim()); //100
        wrapper.find('#btnSubtract').simulate('click'); //-1
        const valueAfterClickSubtract = parseInt(wrapper.find('#idCounter').text().trim()); //99

        const valueCounterNew = (valueAfterClickSubtract + 1);
        expect(valorCounter).toBe(valueCounterNew);

    })

    test('deberia reiniciar el valor por defecto con btn reset', () => {

        const value = wrapper.find('#idCounter').text().trim();

        wrapper.find('#btnSubtract').simulate('click');
        wrapper.find('#btnSubtract').simulate('click');
        wrapper.find('#btnReset').simulate('click');

        const valueAfterModified = wrapper.find('#idCounter').text().trim();
        expect(value).toBe(valueAfterModified);
    })


})
