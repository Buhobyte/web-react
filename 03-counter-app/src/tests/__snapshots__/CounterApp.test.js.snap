// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de CounterApp debe mostrar <CounterApp/> correctamente 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h1>
    0
  </h1>
  <button
    onClick={[Function]}
  >
     +1  
  </button>
  <button
    onClick={[Function]}
  >
     +3 
  </button>
  <button
    onClick={[Function]}
  >
     +2 
  </button>
  <br />
  <h1>
    Ejercicio count
  </h1>
  <h1
    id="idCounter"
  >
    100
  </h1>
  <button
    id="btnPlus"
    onClick={[Function]}
  >
     +1  
  </button>
  <button
    id="btnReset"
    onClick={[Function]}
  >
     Reset  
  </button>
  <button
    id="btnSubtract"
    onClick={[Function]}
  >
     -1  
  </button>
</Fragment>
`;
